projects/batfish/src/main/java/org/batfish/bdp/BdpDataPlanePlugin.java

```diff
@@ -415,7 +416,59 @@ private void computeFixedPoint(
               }
               ospfInternalImportCompleted.incrementAndGet();
             });
-    // END DONE ONCE
+    // END DONE ONCE (OSPF)
+
+    // RIP internal routes
+    final boolean[] ripInternalChanged = new boolean[] {true};
+    int ripInternalIterations = 0;
+    while (ripInternalChanged[0]) {
+      ripInternalIterations++;
+      ripInternalChanged[0] = false;
+      AtomicInteger ripInternalCompleted =
+          _batfish.newBatch(
+              "Compute RIP Internal routes: iteration " + ripInternalIterations, nodes.size());
+      nodes
+          .values()
+          .parallelStream()
+          .forEach(
+              n -> {
+                for (VirtualRouter vr : n._virtualRouters.values()) {
+                  if (vr.propagateRipInternalRoutes(nodes, topology)) {
+                    synchronized (routesChangedMonitor) {
+                      ripInternalChanged[0] = true;
```

Comment:
any reason not to use an AtomicBoolean for this?

Reply:
Didn't know about it at the time. Should do the trick.